### PR TITLE
[PARAM,TOOL] FalseDiscoveryRate filter in tool

### PR DESCRIPTION
**Pull request by timosachsenberg**
- reworked parameters (changed flag to true / false string option with better logic)
- allow FDR control in FDR tool

### CODE REVIEW

**Comment by timosachsenberg** [comment]
adresses https://github.com/OpenMS/OpenMS/issues/2531

File: src/topp/FalseDiscoveryRate.cpp [hunk 1]
```diff
@@ -171,6 +177,20 @@ class TOPPFalseDiscoveryRate :
...
+    if (psm_fdr >= 0)
+    {
+      LOG_INFO << "FDR control: Filtering PSMs..." << endl;
+      IDFilter::filterHitsByScore(pep_ids, psm_fdr / 100.0);
+    }
+
+    double protein_fdr = getDoubleOption_("FDR:protein");
+    if (protein_fdr >= 0)
+    {
+      LOG_INFO << "FDR control: Filtering proteins..." << endl;
```
Comment (timosachsenberg) [inline src/topp/FalseDiscoveryRate.cpp]: I think correct would be to check if the score type is q-value? FDR filtering should work without recalculation if it already contains a q-value. Will add this.

Comment (timosachsenberg) [inline src/topp/FalseDiscoveryRate.cpp]: On the other hand, I think I just filter if FDR has been calculated by the tool.

**Comment by hroest** [comment]
```
The following tests FAILED:

	1781 - TOPP_FalseDiscoveryRate_4_out1 (Failed)

Submit files (using http)
```

**Comment by timosachsenberg** [comment]
call for votes:
1. let the user specify FDR in percent or fraction?
@hroest @jpfeuffer @cbielow @hendrikweisser

**Comment by jpfeuffer** [comment]
I think I would leave it as fraction -> No change and all the statistic tools (e.g. in R) do it, too

**Comment by timosachsenberg** [comment]
agreed. I changed it to fraction and added range constraints.

**Comment by timosachsenberg** [comment]
merge before or after source release?

**Comment by mwalzer** [comment]
I'd say before source release. My rationale would be to, where CLI change is immediate anyway, include as much as possible.

**Comment by hroest** [comment]
also :+1: for fractions, so should we merge?

**Pull request merged**